src/main/java/com/waverleysoftware/listemers/TestListener.java

```diff
@@ -7,24 +7,26 @@
 import org.testng.ITestResult;
 import org.testng.internal.TestResult;
 
-@SuppressWarnings({"JavadocType", "unchecked"})
+@SuppressWarnings("JavadocType")
 @Slf4j
 public class TestListener implements IInvokedMethodListener {
-    private static ThreadLocal thread = new ThreadLocal();
+    private static final ThreadLocal<SoftAssertions> THREAD_LOCAL_CONTAINER_FOR_SOFT_ASSERTIONS = new ThreadLocal<>();
 
     @Override
     public void beforeInvocation(final IInvokedMethod method, final ITestResult testResult) {
         final SoftAssertions softAssert = new SoftAssertions();
-        thread.set(softAssert);
+        THREAD_LOCAL_CONTAINER_FOR_SOFT_ASSERTIONS.set(softAssert);
```

Comment:
no need to create a separate `softAssert` variable... you can inline it with setter